### PR TITLE
Cleanup/initial fixes

### PR DESCRIPTION
**Pull request by charleenfei**
<This is intended as a guide when describing-reviewing pull requests for go-centrifuge>
Issue Link: <Github link>

### CODE REVIEW

File: documents/model.go [hunk 1]
```diff
@@ -74,6 +74,9 @@ type Model interface {
 	// GetCollaborators returns the collaborators of this Document.
 	// filter ids should not be returned
 	GetCollaborators(filterIDs ...identity.CentID) ([]identity.CentID, error)
+
+	// Roles returns the Roles in this Document
+	Roles() []coredocumentpb.Role
```
Comment (vedhavyas) [inline documents/model.go:79]: We should not expose roles. What is the use case ?

File: documents/service.go [hunk 2]
```diff
@@ -253,7 +253,7 @@ func (s service) getVersion(ctx context.Context, documentID, version []byte) (Mo
 }
 
 func (s service) DeriveFromCoreDocument(cd coredocumentpb.CoreDocument) (Model, error) {
-	if cd.EmbeddedData == nil {
+	if cd.EmbeddedData == nil || &cd == nil {
```
Comment (vedhavyas) [inline documents/service.go:256]: `&cd == nil` this is never nil. we dont need to check this

File: testingutils/documents/documents.go [hunk 3]
```diff
@@ -70,33 +74,38 @@ func (m *MockModel) JSON() ([]byte, error) {
...
+	var collabs []string
+	for _, c := range collaborators {
+		id := hexutil.Encode(c)
+		collabs = append(collabs, id)
+	}
+	m, err := documents.NewCoreDocumentWithCollaborators(collabs)
+	if err != nil {
+		return cd, err
+	}
+	invData := &invoicepb.InvoiceData{}
```
Comment (vedhavyas) [inline testingutils/documents/documents.go]:
```suggestion
	invData := new(invoicepb.InvoiceData)
```

**Review by vedhavyas — CHANGES_REQUESTED** [review]
Lets not expose methods on Model Testing. You can use the existing validation methods to do the testing

File: documents/coredocument.go [hunk 4]
```diff
@@ -193,13 +193,13 @@ func (cd *CoreDocument) PrepareNewVersion(collaborators []string, initSalts bool
 	}
 
 	ncd := &CoreDocument{Document: cdp}
-	cd.addCollaboratorsToReadSignRules(ucs)
+	ncd.addCollaboratorsToReadSignRules(ucs)
```
Comment (vedhavyas) [inline documents/coredocument.go:196]: Good catch!

File: documents/model.go (hunk 1 above)
Comment (charleenfei) [reply documents/model.go:79]: in the mintNFTWithProofs test, we want to access the role count from the Model

File: documents/service.go (hunk 2 above)
Comment (charleenfei) [reply documents/service.go:256]: it was returning nil in a test, but I can change the test so it won't check for nil documents

Comment (vedhavyas) [reply documents/service.go:256]: nil most probably due to cd.EmbedData being nil i think

File: testingutils/documents/documents.go [hunk 5]
```diff
@@ -70,33 +74,38 @@ func (m *MockModel) JSON() ([]byte, error) {
...
+	m.Document.EmbeddedDataSalts = documents.ConvertToProtoSalts(dataSalts)
+	cdSalts, _ := documents.GenerateNewSalts(&m.Document, "", nil)
+	m.Document.CoredocumentSalts = documents.ConvertToProtoSalts(cdSalts)
+	cd = *m
+
+	return cd, nil
+}
+
+func GenerateCoreDocumentModel() (documents.CoreDocument, error) {
+	var cd documents.CoreDocument
```
Comment (vedhavyas) [inline testingutils/documents/documents.go:104]: we dont need to initialise cd here. we can return m as is

**Pull request closed without merging**